File: OSSearchExampleObjC/ViewController.m
//
//  ViewController.m
//  SearchExampleObjC
//
//  Created by David Haynes on 17/03/2016.
//  Copyright © 2016 Ordnance Survey. All rights reserved.
//

#import "ViewController.h"
#import "OSResultsViewController.h"

@interface ViewController ()<UISearchResultsUpdating>

@property (nonatomic, strong) UISearchController *searchController;
@property (nonatomic, strong) OSResultsViewController *resultsViewController;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.resultsViewController = [[OSResultsViewController alloc] init];
    self.searchController = [[UISearchController alloc] initWithSearchResultsController:self.resultsViewController];
    self.searchController.searchResultsUpdater = self;
    self.searchController.dimsBackgroundDuringPresentation = YES;
    self.searchController.searchBar.scopeButtonTitles = @[ @"OS Places", @"OS Open Names" ];
    [self.view addSubview:self.searchController.searchBar];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

- (void)updateSearchResultsForSearchController:(UISearchController *)searchController {
    [self.resultsViewController search:searchController.searchBar.text scope:searchController.searchBar.selectedScopeButtonIndex];
}

- (BOOL)prefersStatusBarHidden {
    return YES;
}

@end

File: OSSearchExampleSwift/ViewController.swift
//
//  ViewController.swift
//  SearchExampleSwift
//
//  Created by David Haynes on 10/03/2016.
//  Copyright © 2016 Ordnance Survey. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UISearchResultsUpdating {

    var searchController: UISearchController
    var resultsController: ResultsViewController

    required init?(coder aDecoder: NSCoder) {
        self.resultsController = ResultsViewController()
        self.searchController = UISearchController(searchResultsController: resultsController)
        super.init(coder: aDecoder)
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        searchController.searchResultsUpdater = self
        searchController.dimsBackgroundDuringPresentation = true
        definesPresentationContext = true
        searchController.searchBar.scopeButtonTitles = ["OS Places", "OS Open Names"]
        self.view.addSubview(searchController.searchBar)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    func updateSearchResultsForSearchController(searchController: UISearchController) {
        if let text = searchController.searchBar.text, scope = SearchScope(rawValue: searchController.searchBar.selectedScopeButtonIndex) {
            resultsController.search(text, scope: scope)
        }
    }

    override func prefersStatusBarHidden() -> Bool {
        return true
    }

}
